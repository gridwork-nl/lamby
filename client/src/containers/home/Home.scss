
.home {
  background: url('../../img/splatter-dark.svg');
  background-position: center;
  background-size: contain;
  background-repeat: no-repeat;
}
