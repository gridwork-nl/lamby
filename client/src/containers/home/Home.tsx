import './Home.scss'
import { useContext } from 'react'
import { Link } from 'react-router-dom';
import Timer from '../../components/timer/Timer'
import { AppStateContext } from '../../context/AppContext'

const Home: React.FC = () => {
  const { data } = useContext(AppStateContext)
  const match = data.match

  return (
    <div className='home-wrapper'>
      <header className='header'>
        <h1>HOME</h1>
      </header>
      <div className='home center-div'>
        <h1>{match.homeTeam}</h1>
        <h4>vs</h4>
        <h1>{match.awayTeam}</h1>
        <h4>{match.venue}</h4>
        {match && <Timer />}
        <Link to='/match/edit' className='btn main-btn'>EDIT MATCH</Link>
      </div>
    </div>
  );
}

export default Home
