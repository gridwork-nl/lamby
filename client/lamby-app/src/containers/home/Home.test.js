import React from 'react';
import ReactDom from 'react-dom';
import {
  cleanup,
  getByTestId,
  render,
  fireEvent,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ShallowRenderer from 'react-test-renderer/shallow';
import TestRenderer from 'react-test-renderer';
import Home from './Home';

afterEach(cleanup);
it('renders correctly Home-Page', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Home />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
