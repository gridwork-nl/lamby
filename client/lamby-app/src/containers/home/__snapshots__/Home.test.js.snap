// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Home-Page 1`] = `
<div
  className="home-wrapper"
>
  <header
    className="header"
  >
    <h1>
      HOME
    </h1>
  </header>
  <div
    className="home center-div"
  >
    <h1>
      
    </h1>
    <h4>
      vs
    </h4>
    <h1>
      
    </h1>
    <h4>
      
    </h4>
    <Timer />
    <Link
      className="btn main-btn"
      to="/match/edit"
    >
      EDIT MATCH
    </Link>
  </div>
</div>
`;
